### DISCUSSION TITLE
RabbitMQ Quorum Queue (QQ) and Priority Queues (PQ)

### DISCUSSION THREAD

**Discussion by CrowSoda** [rabbitmq/rabbitmq-server#12997] [Ideas] [answered]
Howdy all, I am trying to get the functionality of both QQ and PQ. I've been scouring the documentation and have pretty much come to the conclusion that it isn't possible without doing something super hacky, has anyone else looked into this?

Here are workarounds that I have found or things I looked into an exhausted:

- Define separate queues and handle the logic at the worker.
- Handle all the priority logic at the worker level.
- [Use the "Redis" method (untested).](https://docs.celeryq.dev/en/latest/userguide/routing.html#redis-message-priorities)
- I have not tried to force a priority by declaring it.

[Also, poking around here](https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbit/src/rabbit_quorum_queue.erl), I have found that it only explicitly says that `<<"max-priority">>` (Lines 519-525) aren't supported and then on lines 985-990, it appears that I can still assign a priority (I haven't gone deep into this file, so my surface level assumption might be wrong).

Does the lack of `max-priority` make it impossible to do a priority queue?

**Comment by lukebakken** [comment, marked as answer]
Please see the documentation:

https://www.rabbitmq.com/docs/quorum-queues#priorities

**Reply by CrowSoda** [reply]
How on earth did I miss that, I am honestly at a loss. Thanks Luke.

**Reply by lukebakken** [reply]
No worries! Happy to help out.
